confmap/internal/e2e/types_test.go

```diff
@@ -98,27 +98,22 @@ func TestTypeCasting(t *testing.T) {
 		{
 			value:       "\"0123\"",
 			targetField: TargetFieldString,
-			expected:    "0123",
-		},
-		{
-			value:       "\"0123\"",
-			targetField: TargetFieldInt,
-			expected:    83,
+			expected:    "\"0123\"",
 		},
 		{
 			value:       "\"0123\"",
 			targetField: TargetFieldInlineString,
-			expected:    "inline field with 0123 expansion",
+			expected:    "inline field with \"0123\" expansion",
 		},
 		{
 			value:       "!!str 0123",
 			targetField: TargetFieldString,
-			expected:    "0123",
+			expected:    "!!str 0123",
```

Comment:
This only applies on provider expanded strings correct? QBQ it isn't breaking all yaml rules for everything :D

Reply:
Correct, only when expanding using the env provider, which happens with `${env:}` and `${}`